frontend/src/app/modules/matches/matches-computations.js

```diff
@@ -68,3 +77,104 @@ export const computeWinRatio = (playerId, playerMatches) => {
   const wonMatchesCount = playerMatches.filter(match => didPlayerWin(playerId, match)).length
   return (playerMatches.length > 0) ? (wonMatchesCount / playerMatches.length) : 0
 }
+
+export const computeDays = matchChanges => {
+  let worstDayKey = ''
+  let bestDayKey = ''
+  const daysMap = {}
+  for (const match of matchChanges) {
+    const key = match.date.toLocaleDateString()
+    const value = Number(match.ratingChangeString)
+    if (!Object.hasOwnProperty.call(daysMap, key)) {
+      daysMap[key] = {
+        date: key,
+        value,
+      }
+    } else {
+      daysMap[key]['value'] += value
+    }
+
+    bestDayKey = updateMaxFromMap(daysMap[bestDayKey] ?? {}, 'value', daysMap[key]).date
+    worstDayKey = updateMinFromMap(daysMap[worstDayKey] ?? {}, 'value', daysMap[key]).date
+  }
+
+  return {
+    bestDay: daysMap[bestDayKey],
```

Comment:
we may return empty object if there are no matches, is it expected?

Reply:
Fixed by providing default values [here](https://github.com/salsita/foosball-rating/pull/114/commits/3d398530cdf75f980e6344e885e4f18dd919311a#diff-304dca28d92d292252e86b7e83fc1569)